tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -395,9 +398,353 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+			gomega.Expect(e2eVSphere.verifyDatastoreMatch(volumeID, sharedvmfsURL)).To(
+				gomega.BeTrue(), "volume %v was created on wrong ds", sharedvmfsURL)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			for i, pvc := range pvcs {
+				err := fpv.DeletePersistentVolumeClaim(client, pvc.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(pvs[i].Spec.CSI.VolumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Create pods with using the PVCs created in step 3 and wait for them to be ready")
+		ginkgo.By("verify we can read and write on the PVCs")
+		pods := createMultiplePods(ctx, client, pvclaims2d, true)
+		defer func() {
+			ginkgo.By("Delete pods")
+			deletePodsAndWaitForVolsToDetach(ctx, client, pods, true)
+		}()
+
+		rand.Seed(time.Now().Unix())
+		testdataFile := fmt.Sprintf("/tmp/testdata_%v_%v", time.Now().Unix(), rand.Intn(1000))
+		ginkgo.By(fmt.Sprintf("Creating a 512mb test data file %v", testdataFile))
+		op, err := exec.Command("dd", "if=/dev/urandom", fmt.Sprintf("of=%v", testdataFile),
+			"bs=64k", "count=8000").Output()
+		fmt.Println(op)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			op, err = exec.Command("rm", "-f", testdataFile).Output()
+			fmt.Println(op)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expand pvcs while writing some data on them")
+		var wg sync.WaitGroup
+		currentPvcSize := pvcs[0].Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Add(resource.MustParse("2Gi"))
+		fsSizes := []int64{}
+		for _, pod := range pods {
+			originalSizeInMb, err := getFSSizeMb(f, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			fsSizes = append(fsSizes, originalSizeInMb)
+		}
+		wg.Add(4)
+		go copyFile2PodInParallel(pods[0], testdataFile, &wg)
+		go copyFile2PodInParallel(pods[1], testdataFile, &wg)
+		go resize(client, pvcs[0], pvcs[0].Spec.Resources.Requests[v1.ResourceStorage], newSize, &wg)
+		go resize(client, pvcs[1], pvcs[1].Spec.Resources.Requests[v1.ResourceStorage], newSize, &wg)
+		wg.Wait()
+
+		ginkgo.By("Wait and verify the file system resize on pvcs")
+		for i := range pvcs {
+			framework.Logf("Waiting for file system resize to finish for pvc %v", pvcs[i].Name)
```

Comment:
We should also check for pv resize here

Reply:
Verifying FS resize implicitly means that PV resize is also done.